agent/test-agent-cli/src/client.rs

```diff
@@ -0,0 +1,119 @@
+use crate::error::{self, K8sSnafu, ResolveConfigSnafu, ResourceClientCreateSnafu, Result};
```

Comment:
How does this file vary from `test-agent/src/k8s_client.rs`?

Reply:
This client doen't implement any trait, its a struct by which we talk to k8s client. Some of the methods and return types are different too. The Configuration type in Spec is also different.

Reply:
Is there a way we can use the existing k8s_client (with some small changes). There seems to be a lot of duplicated logic.